workflows/diploshic.snake

```diff
@@ -0,0 +1,164 @@
+
+"""
+Snakefile module for training, and eventually running,
+a diploSHIC classifier on stdpopsim simulations.
+
+"""
+
+import pathlib
+import sys
+import numpy as np
+import yaml
+
+seed = 12345
+np.random.seed(seed)
+replicates = 2
+seed_array = np.random.random_integers(1,2**31,replicates)
+
+# change to the correct locations of discoal and diploSHIC
+discoal_exec = "ext/discoal/discoal"
+diploSHIC_exec = "diploSHIC"
+
+#some parameters that live in a .yaml file
+with open('workflows/diploshic_params.yaml') as file:
+      params = yaml.load(file, Loader=yaml.FullLoader)
+trainSampleNumber = params['trainSampleNumber'] #the number of simulation replicates we want to generate for each file in our training set
+testSampleNumber = params['testSampleNumber'] #the number of simulations to create for each file in the test set
+sampleSize = params['sampleSize'] #the number of individuals in our population sample
+numSites = params['numSites'] #total number of sites in our simulated window (i.e. S/HIC's subwindow size * 11)
+u = params['u'] #per-site mutation rate (used to calculate mean theta)
+gensPerYear = params['gensPerYear'] #number of generations per year
+maxSoftSweepInitFreq = params['maxSoftSweepInitFreq'] #maximum initial selected frequency for soft sweeps
+tauHigh = params['tauHigh'] #maximum FIXATION (not mutation) time (in units of 4N generations ago) in the past
+rhoOverTheta = params['rhoOverTheta'] #crossover rate over mut rate (used to calculate mean rho)
+ne0=params['ne0'] #effective population size at present
+N0 = ne0
+thetaMean=4*N0*float(u)*numSites
+rhoMean = thetaMean * rhoOverTheta
+thetaLow = (2*thetaMean)/11.0
+thetaHigh = 10*thetaLow
+rhoMax = 3 * rhoMean
+
+totSimRuns = params['totSimRuns'] #total number of simulations to run (i.e. trainSampleNumber + testSampleNumber)
+
+neutString = f"{discoal_exec} {sampleSize} {testSampleNumber} {numSites} -Pt {thetaLow} {thetaHigh} -Pre {rhoMean} {rhoMax} "
+# TODO add demography string to the neutral string above
```

Comment:
Is this TODO necessary, or can we keep training to equilibrium demography?

Reply:
yeah not necessary i guess for what we want to do here